### DISCUSSION TITLE
Different results from microeco 0.14 and 0.19.5

### DISCUSSION THREAD

**Discussion by amrvx4** [ChiLiubio/microeco#244] [Q&A]
I ran analysis on the same data recently using 0.19.5 version. I previously analyzed that data using 0.14... version. I find the results I got are different. I am attaching here both figures for example from old and new version. The data is exact same because it was a phyloseq I saved which was just converted via file2meco.

Old version:
![OLD_Version](https://github.com/ChiLiubio/microeco/assets/101592620/834dee6e-18f1-42fb-964b-e3f5b204eea8)

New version: 
![New_version](https://github.com/ChiLiubio/microeco/assets/101592620/b8ffd7af-c18d-4200-bba3-500c714173f3)

I am not sure why when I analyze it in new version I dont even see the names of corresponding phylum for each clade. Is there any change in the analysis of LEFse? or Is that any added feature to show or hide phylum and other taxa?

Thanks,

**Comment by ChiLiubio** [comment]
Hi @amrvx4 ,
I carefully checked the codes and history commits between v0.14.0 and v0.19.5. The core steps are same. I use the dataset inside the package to run same steps as the dataset in the package has no change between two versions. The following is what I test. The results are also same. The minor changes probably come from the randomizatons in the algorithms, which has been discussed in a previous Issue.

```
# v0.14.0
rm(list = ls())
library(microeco)
library(magrittr)
data(dataset)
dataset$sample_table %<>% .[.$Group %in% c("CW", "IW"), ]
dataset$tidy_dataset()
dataset$cal_abund()
dataset$save_abund("taxa_abund_v0.14.0")
t3 <- trans_diff$new(dataset = dataset, method = "lefse", group = "Group", alpha = 0.05)
write.csv(t3$res_diff, "lefse_0.14.0.csv")
g1 <- t3$plot_diff_bar(use_number = 1:30)
ggplot2::ggsave("lefse_bar_0.14.0.png", g1, width = 10, height= 8)
g2 <- t3$plot_diff_cladogram(use_taxa_num = 200, use_feature_num = 30, clade_label_level = 5)
ggplot2::ggsave("lefse_cladogram_0.14.0.png", g2, width = 10, height= 10)

# v0.19.5
rm(list = ls())
library(microeco)
library(magrittr)
data(dataset)
dataset$sample_table %<>% .[.$Group %in% c("CW", "IW"), ]
dataset$tidy_dataset()
dataset$cal_abund()
dataset$save_abund("taxa_abund_v0.19.5")
t3 <- trans_diff$new(dataset = dataset, method = "lefse", group = "Group", alpha = 0.05)
write.csv(t3$res_diff, "lefse_0.19.5.csv")
g1 <- t3$plot_diff_bar(use_number = 1:30)
ggplot2::ggsave("lefse_bar_0.19.5.png", g1, width = 10, height= 8)
g2 <- t3$plot_diff_cladogram(use_taxa_num = 200, use_feature_num = 30, clade_label_level = 5)
ggplot2::ggsave("lefse_cladogram_0.19.5.png", g2, width = 10, height= 10)
```
So please first check your data. If you are not sure how this happens, please send me your saved data to check it following the tutorial (https://chiliubio.github.io/microeco_tutorial/notes.html#save-function).
